client/src/components/HistoryViewer/HistoryViewerVersionDetail.js

```diff
@@ -173,45 +151,56 @@ class HistoryViewerVersionDetail extends PureComponent {
     const extraListClasses = {
       'history-viewer__table': true,
       'history-viewer__table--current': true,
-      'history-viewer__table--compare': this.isCompareMode(),
+      'history-viewer__table--compare': isCompareMode(),
     };
     const formClasses = {
       'history-viewer__version-detail': true,
-      'history-viewer__version-detail--compare': this.isCompareMode(),
+      'history-viewer__version-detail--compare': isCompareMode(),
     };
     return (
       <div className="flexbox-area-grow fill-height">
         <CompareWarningComponent fixed />
-
         <div className={classnames(containerClasses)}>
           <ListComponent
             extraClass={classnames(extraListClasses)}
-            versions={this.getListVersions()}
+            versions={getListVersions()}
             compareModeAvailable={compareModeAvailable}
           />
-
           <div className={classnames(formClasses)}>
             <FormBuilderLoaderComponent
               identifier="HistoryViewer.VersionDetail"
               schemaUrl={schemaUrl}
             />
           </div>
         </div>
-
-        {this.renderToolbar()}
+        {renderToolbar()}
       </div>
     );
-  }
+  };
 
-  render() {
-    return (
-      <div className="flexbox-area-grow fill-width">
-        {this.renderDetails()}
-        {this.renderPreview()}
-      </div>
-    );
-  }
-}
+  /**
+   * When new props are received (from Redux dispatch events), check whether the preview
+   * state changes. If so, we want to add or remove the legacy CSS modifier for the CMS
+   * north toolbar based on whether the view mode is "split" (add) or anything else (remove)
+   */
```

Comment:
This comment should be on the other `useEffect` - this one is for mounting, not updating.

Reply:
Updated